chapter2.adoc

```diff
@@ -65,6 +65,9 @@ Debug operations in all modes are prohibited for hart i when the logical-OR of `
 | 0      | 0      | n/a                      
 |============================================
 
+==== Privilege Changing Instructions
+
+The RISC-V Debug Specification defines that the instructions that change the privilege mode have UNSPECIFIED behavior when executed within the Program Buffer, with exception of the ebreak instruction. Those instructions such as mret, sret, uret, ecall, may either act as NOP or trigger an exception, stopping execution and setting `cmderr` to 3. Notably, these instruction retain their functionality during single stepping.
```

Comment:
This looks like it addresses the point I originally raised.  I have some minor wording concerns.

The first sentence relates to another spec.  The second sentence needs to be clearer that it is specifying a requirement in _this_ spec rather than also restating something about the other spec.  I would also replace "may" with "must" in the second sentence since it's a requirement that one of the two behaviors happens.

I propose this:
```suggestion
The RISC-V Debug Specification defines that the instructions that change the privilege mode have UNSPECIFIED behavior when executed within the Program Buffer, with exception of the ebreak instruction. In Sdsec, those instructions such as mret, sret, uret, ecall, must either act as NOP or trigger an exception, stopping execution and setting `cmderr` to 3. Notably, these instructions retain their normal functionality during single stepping.
```

Reply:
Thanks for the suggestion! Agree on you you point and committing the suggestion.